In [1]:
import pandas as pd

In [2]:
def process(ckpt):
    ret_data = {'public_test': {}, 'private_test': {}}
    
    info, _, public_test_data = ckpt[0].split('**')
    name, public_per = info.split('[public_test] PER: ')
    public_per, private_per = public_per.split('[private_test] PER: ')
    name = name.split('Checkpoint: ')[-1]

    ret_data['public_test']['PER'] = float(public_per.strip())
    ret_data['private_test']['PER'] = float(private_per.strip())

    key_data = public_test_data.strip().split('\n')
    key_data = [i.split(': ') for i in key_data]
    public_test_metrics = {k: float(v.split('|')[0]) for k, v in key_data}

    ret_data['public_test'].update(public_test_metrics)

    key_data = ckpt[1].strip().split('\n')[1:]
    key_data = [i.split(': ') for i in key_data]
    private_test_metrics = {k: float(v.split('|')[0]) for k, v in key_data}

    ret_data['private_test'].update(private_test_metrics)

    return {'name': name.strip(), 'metrics': ret_data}

In [3]:
def analyze_result(result_file):
    data = open(result_file).read()
    per_ckpt = data.split('================')
    per_ckpt = [i for i in per_ckpt if i.strip()]
    per_ckpt = [i.strip().split('\n\n\n') for i in per_ckpt]
    process_data = list(map(process, per_ckpt))
    public_metrics = {i['name']: i['metrics']['public_test'] for i in process_data}
    public_df = pd.DataFrame(public_metrics).T
    private_metrics = {i['name']: i['metrics']['private_test'] for i in process_data}
    private_df = pd.DataFrame(private_metrics).T

    return public_df, private_df

In [4]:
public_df, private_df = analyze_result('ablation_study/spec_augment')

In [5]:
public_df, private_df = analyze_result('ablation_study/with_linguistic_head')

In [6]:
public_df

PER  Correct Rate  \
w2v2_ablation_with_ling_head-best_on_tp0.025_tl...  0.1647        0.9529   
w2v2_ablation_with_ling_head-best_on_tp0.025_tl...  0.0798        0.9547   
w2v2_ablation_with_ling_head-drop0.1-not-load-b...  0.0909        0.9570   
w2v2_ablation_with_ling_head-drop0.05-not-load-...  0.0925        0.9560   
w2v2_ablation_with_ling_head-0drop-load-best-pe...  0.1718        0.9464   
w2v2_ablation_200epoch-with_ling_head-0drop-0lo...  0.1539        0.9579   
w2v2_ablation_200epoch-with_ling_head-0.1drop-0...  0.0805        0.9621   

                                                    Accuracy  Recall  \
w2v2_ablation_with_ling_head-best_on_tp0.025_tl...    0.8353  0.6647   
w2v2_ablation_with_ling_head-best_on_tp0.025_tl...    0.9202  0.4869   
w2v2_ablation_with_ling_head-drop0.1-not-load-b...    0.9091  0.4402   
w2v2_ablation_with_ling_head-drop0.05-not-load-...    0.9075  0.4606   
w2v2_ablation_with_ling_head-0drop-load-best-pe...    0.8282  0.6152   
w2v2_ablation_200epoch-with_ling_head-0drop-0lo...    0.8461  0.4985   
w2v2_ablation_200epoch-with_ling_head-0.1drop-0...    0.9195  0.4052   

                                                    Precision  f1-score  \
w2v2_ablation_with_ling_head-best_on_tp0.025_tl...     0.3945    0.4951   
w2v2_ablation_with_ling_head-best_on_tp0.025_tl...     0.3568    0.4118   
w2v2_ablation_with_ling_head-drop0.1-not-load-b...     0.3604    0.3963   
w2v2_ablation_with_ling_head-drop0.05-not-load-...     0.3575    0.4025   
w2v2_ablation_with_ling_head-0drop-load-best-pe...     0.3381    0.4364   
w2v2_ablation_200epoch-with_ling_head-0drop-0lo...     0.4043    0.4465   
w2v2_ablation_200epoch-with_ling_head-0.1drop-0...     0.4149    0.4100   

                                                    True Acception  \
w2v2_ablation_with_ling_head-best_on_tp0.025_tl...          0.9700   
w2v2_ablation_with_ling_head-best_on_tp0.025_tl...          0.9742   
w2v2_ablation_with_ling_head-drop0.1-not-load-b...          0.9770   
w2v2_ablation_with_ling_head-drop0.05-not-load-...          0.9757   
w2v2_ablation_with_ling_head-0drop-load-best-pe...          0.9646   
w2v2_ablation_200epoch-with_ling_head-0drop-0lo...          0.9784   
w2v2_ablation_200epoch-with_ling_head-0.1drop-0...          0.9832   

                                                    False Rejection  \
w2v2_ablation_with_ling_head-best_on_tp0.025_tl...           0.0300   
w2v2_ablation_with_ling_head-best_on_tp0.025_tl...           0.0258   
w2v2_ablation_with_ling_head-drop0.1-not-load-b...           0.0230   
w2v2_ablation_with_ling_head-drop0.05-not-load-...           0.0243   
w2v2_ablation_with_ling_head-0drop-load-best-pe...           0.0354   
w2v2_ablation_200epoch-with_ling_head-0drop-0lo...           0.0216   
w2v2_ablation_200epoch-with_ling_head-0.1drop-0...           0.0168   

                                                    False Acceptance  \
w2v2_ablation_with_ling_head-best_on_tp0.025_tl...            0.3353   
w2v2_ablation_with_ling_head-best_on_tp0.025_tl...            0.5131   
w2v2_ablation_with_ling_head-drop0.1-not-load-b...            0.5598   
w2v2_ablation_with_ling_head-drop0.05-not-load-...            0.5394   
w2v2_ablation_with_ling_head-0drop-load-best-pe...            0.3848   
w2v2_ablation_200epoch-with_ling_head-0drop-0lo...            0.5015   
w2v2_ablation_200epoch-with_ling_head-0.1drop-0...            0.5948   

                                                    Correct Diagnosis  \
w2v2_ablation_with_ling_head-best_on_tp0.025_tl...             0.5263   
w2v2_ablation_with_ling_head-best_on_tp0.025_tl...             0.5629   
w2v2_ablation_with_ling_head-drop0.1-not-load-b...             0.5762   
w2v2_ablation_with_ling_head-drop0.05-not-load-...             0.5759   
w2v2_ablation_with_ling_head-0drop-load-best-pe...             0.4787   
w2v2_ablation_200epoch-with_ling_head-0drop-0lo...             0.4561   
w2v2_ablation_200epoch-with_ling_head-0.1drop-0...

In [7]:
private_df

PER  Correct Rate  \
w2v2_ablation_with_ling_head-best_on_tp0.025_tl...  0.2075        0.9127   
w2v2_ablation_with_ling_head-best_on_tp0.025_tl...  0.1324        0.9126   
w2v2_ablation_with_ling_head-drop0.1-not-load-b...  0.1454        0.9169   
w2v2_ablation_with_ling_head-drop0.05-not-load-...  0.1427        0.9185   
w2v2_ablation_with_ling_head-0drop-load-best-pe...  0.2254        0.9019   
w2v2_ablation_200epoch-with_ling_head-0drop-0lo...  0.2094        0.9194   
w2v2_ablation_200epoch-with_ling_head-0.1drop-0...  0.1360        0.9224   

                                                    Accuracy  Recall  \
w2v2_ablation_with_ling_head-best_on_tp0.025_tl...    0.7925  0.7268   
w2v2_ablation_with_ling_head-best_on_tp0.025_tl...    0.8676  0.6286   
w2v2_ablation_with_ling_head-drop0.1-not-load-b...    0.8545  0.6134   
w2v2_ablation_with_ling_head-drop0.05-not-load-...    0.8573  0.6196   
w2v2_ablation_with_ling_head-0drop-load-best-pe...    0.7746  0.7357   
w2v2_ablation_200epoch-with_ling_head-0drop-0lo...    0.7905  0.6420   
w2v2_ablation_200epoch-with_ling_head-0.1drop-0...    0.8640  0.5732   

                                                    Precision  f1-score  \
w2v2_ablation_with_ling_head-best_on_tp0.025_tl...     0.5062    0.5968   
w2v2_ablation_with_ling_head-best_on_tp0.025_tl...     0.4927    0.5524   
w2v2_ablation_with_ling_head-drop0.1-not-load-b...     0.5131    0.5588   
w2v2_ablation_with_ling_head-drop0.05-not-load-...     0.5195    0.5651   
w2v2_ablation_with_ling_head-0drop-load-best-pe...     0.4666    0.5710   
w2v2_ablation_200epoch-with_ling_head-0drop-0lo...     0.5346    0.5834   
w2v2_ablation_200epoch-with_ling_head-0.1drop-0...     0.5445    0.5585   

                                                    True Acception  \
w2v2_ablation_with_ling_head-best_on_tp0.025_tl...          0.9423   
w2v2_ablation_with_ling_head-best_on_tp0.025_tl...          0.9473   
w2v2_ablation_with_ling_head-drop0.1-not-load-b...          0.9526   
w2v2_ablation_with_ling_head-drop0.05-not-load-...          0.9533   
w2v2_ablation_with_ling_head-0drop-load-best-pe...          0.9315   
w2v2_ablation_200epoch-with_ling_head-0drop-0lo...          0.9545   
w2v2_ablation_200epoch-with_ling_head-0.1drop-0...          0.9610   

                                                    False Rejection  \
w2v2_ablation_with_ling_head-best_on_tp0.025_tl...           0.0577   
w2v2_ablation_with_ling_head-best_on_tp0.025_tl...           0.0527   
w2v2_ablation_with_ling_head-drop0.1-not-load-b...           0.0474   
w2v2_ablation_with_ling_head-drop0.05-not-load-...           0.0467   
w2v2_ablation_with_ling_head-0drop-load-best-pe...           0.0685   
w2v2_ablation_200epoch-with_ling_head-0drop-0lo...           0.0455   
w2v2_ablation_200epoch-with_ling_head-0.1drop-0...           0.0390   

                                                    False Acceptance  \
w2v2_ablation_with_ling_head-best_on_tp0.025_tl...            0.2732   
w2v2_ablation_with_ling_head-best_on_tp0.025_tl...            0.3714   
w2v2_ablation_with_ling_head-drop0.1-not-load-b...            0.3866   
w2v2_ablation_with_ling_head-drop0.05-not-load-...            0.3804   
w2v2_ablation_with_ling_head-0drop-load-best-pe...            0.2643   
w2v2_ablation_200epoch-with_ling_head-0drop-0lo...            0.3580   
w2v2_ablation_200epoch-with_ling_head-0.1drop-0...            0.4268   

                                                    Correct Diagnosis  \
w2v2_ablation_with_ling_head-best_on_tp0.025_tl...             0.6708   
w2v2_ablation_with_ling_head-best_on_tp0.025_tl...             0.6989   
w2v2_ablation_with_ling_head-drop0.1-not-load-b...             0.6798   
w2v2_ablation_with_ling_head-drop0.05-not-load-...             0.7133   
w2v2_ablation_with_ling_head-0drop-load-best-pe...             0.6578   
w2v2_ablation_200epoch-with_ling_head-0drop-0lo...             0.6259   
w2v2_ablation_200epoch-with_ling_head-0.1drop-0...

1. 0drop, 0load 20.75 | 59.68
2. 0.1drop, load 13.24 | 55.24
3. 0.1drop, 0load 14.54 | 55.88
4. 0.05drop, 0load 14.27 | 56.51
5. 0drop, load 22.54 | 57.1
6. 200, 0drop, 0load 20.94 | 58.34
7. 200, 0.1drop, 0load 13.60 | 55.85

1. 0drop, 0load 20.75 | 0.7268	0.5062	0.5968
4. 0.05drop, 0load 14.27 | 56.51
3. 0.1drop, 0load 14.54 | 55.88

6. 200, 0drop, 0load 20.94 | 58.34
7. 200, 0.1drop, 0load 13.60 | 55.85

In [ ]:
0.1454	0.6134	0.5131	0.5588	